specification/azurelargeinstance/resource-manager/Microsoft.AzureLargeInstance/preview/2023-07-20-preview/azurelargeinstance.json

```diff
@@ -0,0 +1,1106 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-07-20-preview",
+    "title": "Azure Large Instance",
+    "description": "The AzureLargeInstance Management client",
+    "x-ms-code-generation-settings": {
+      "name": "AzureLargeInstanceClient"
+    }
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "tags": [
+    {
+      "name": "Azure Large Instances",
+      "description": "Handle Operations for Compute Azure Large Instances."
+    },
+    {
+      "name": "Azure Large Storage Instances",
+      "description": "Handle Operations for Storage Azure Large Instances."
+    },
+    {
+      "name": "Common",
+      "description": "Handle Common Operations across Azure Large Instances (Compute, and Storage)"
+    }
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AzureLargeInstance/azureLargeInstance": {
```

Comment:
>azureLargeInstance [](http://example.com/codeflow?start=76&length=18)

The resource type should be plural i.e., here it should be azureLargeInstances?

Also, why not just name the resource type as "instances" as AzureLargeInstance is already indicate the kind of instances.

Reply:
Thanks! I'll update the plural. The reason we don't call it resources is because under the Microsoft.AzureLargeInstance provider we have two different types of instances: Storage (azureLargeStorageInstance) and Compute (azureLargeInstance).